src/main/scala/chisel3/util/BitPat.scala

```diff
@@ -32,7 +32,7 @@ object BitPat {
         bits = (bits << 1) + (if (d == '1') 1 else 0)
       }
     }
-    (bits, mask, x.length - 1)
+    (bits, mask, x.filter(_!='_').length - 1)
```

Comment:
I think this is structurally brittle, some it's duplicating the ignore-character logic with the main loop.

The current structure isn't great (lots of use of mutable state), but I think the most consistent way to write it would be to keep a mutable count variable that increments for each character, and use that in the return?

Alternative, you can rewrite everything in a more functional style, but that's probably a bit excessive.